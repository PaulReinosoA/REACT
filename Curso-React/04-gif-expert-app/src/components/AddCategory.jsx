import { useState } from 'react';
//SI EL LOG SE LIMPIA POR LA EJECUCION ESTE PUEDE CONFIGURARASE PARA PRECERVARLO EN SETINGS DE f12 PRECERVLOG
//export const AddCategory = ( {setcategories} ) => {
export const AddCategory = ( {onSendNewCategory} ) => {

    const [inputstateValue, setInputState] = useState()
    
    const onImputChange = ({target})=>{ //desestructuro del event el target
        //detecta el cambio y lo pone en inputstateValue
        setInputState(target.value);
    }

    const onSubmit = (event)=>{

        event.preventDefault();
        if(inputstateValue.trim().length <=1 ) return //valido no insertar vacios
        if(inputstateValue.trim().length <=1 ) return
        //const categoryUno = inputstateValue;        
        //setcategories( categ => [categoryUno, ...categ] );//se avisa sobre el cambio a redibujar del componente padre
        onSendNewCategory( inputstateValue.trim() );//se avisa sobre el cambio a redibujar del componente padre
        setInputState('')//limpio el input tras cada enter
    }
    
    return (
        <form onSubmit={ (event)=>{onSubmit(event)}/*SUBMIT - ENTER*/ }>
            <input type="text"
                placeholder="Buscar Gifs"
                value={inputstateValue}
                //onChange={ (event)=> {onImputChange(event)} }
                onChange={ onImputChange } //abrebiatura
            />
        </form>
    )
}

