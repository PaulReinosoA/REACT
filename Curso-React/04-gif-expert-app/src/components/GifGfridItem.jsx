import PropTypes from 'prop-types';
import React from 'react'

export const GifGfridItem = ({id,url,title}) => {
    return (
        <div className="card">
                <img src={url} alt={title}/>                
                <p>{title}</p>
        </div>
    )
}

GifGfridItem.propTypes={
    url:PropTypes.string.isRequired,
    title:PropTypes.string.isRequired,
}