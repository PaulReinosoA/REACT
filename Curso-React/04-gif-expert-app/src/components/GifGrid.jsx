import React from 'react'

export const GifGrid = ({category}) => {
  
    const getGifs = async ()=>{

        const url =`https://api.giphy.com/v1/gifs/trending?key=oJHDZPbqIEavBSoj37exjyFUzbE48RL7&q=${category}&limit=20`;
        const resp = await fetch( url )
        const { data } = await resp.json();
        
        const gifs = data.map(img=>{
          return{
            id:img.id,
            title:img.title,
            url:img.images.downsized_medium.url
          }  
        })
        console.log(gifs);
    }
  
    getGifs();//no se hace

    return (
    <>
        <h3>{category}</h3>
        <p>Hol mundo</p>
    </>
  )
}
