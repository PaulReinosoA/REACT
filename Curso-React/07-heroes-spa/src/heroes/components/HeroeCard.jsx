/* eslint-disable no-dupe-keys */
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export const HeroeCard = ({
  id,
  superhero,
  publisher,
  alter_ego: alterEgo,
  first_appearance: firstAppearance,
  characters,
}) => {
  const heroImage = `../src/assets/heroes/${id}.jpg`;
  // animaciones link https://animate.style/

  return (
    <>
      <div className="col animate__animated animate__fadeIn">
        <div className="card">
          <div className="row no-gutters">
            <div className="col-4">
              <img src={heroImage} alt={superhero} className="card-img" />
            </div>
            <div className="col-8">
              <div className="card-body">
                <h5 className="card-title">{superhero}</h5>
                <p className="card-text">{alterEgo}</p>
                {alterEgo !== characters && (
                  <p className="card-text">{characters}</p>
                )}
                <p className="card-text">
                  <small className="text-muted">{firstAppearance}</small>
                </p>
                <p className="card-text">
                  <small className="text-muted">{publisher}</small>
                </p>
                <Link to={`/hero/${id}`}>Mas...</Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

HeroeCard.propTypes = {
  publisher: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  superhero: PropTypes.string.isRequired,
  publisher: PropTypes.string.isRequired,
  alter_ego: PropTypes.string.isRequired,
  first_appearance: PropTypes.string.isRequired,
  characters: PropTypes.string.isRequired,
};
