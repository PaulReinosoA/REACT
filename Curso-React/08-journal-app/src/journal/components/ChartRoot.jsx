import * as am5 from '@amcharts/amcharts5';
import * as am5xy from '@amcharts/amcharts5/xy';
import am5themesAnimated from '@amcharts/amcharts5/themes/Animated';
import { useLayoutEffect } from 'react';
import { PropTypes } from 'prop-types';
import { Grid } from '@mui/material';

export const ChartRoot = ({ idGrid }) => {
  useLayoutEffect(() => {
    const root = am5.Root.new(idGrid);
    root.setThemes([am5themesAnimated.new(root)]);

    const chart = root.container.children.push(
      am5xy.XYChart.new(root, {
        panX: false,
        panY: false,
        wheelX: 'panX',
        wheelY: 'zoomX',
        paddingLeft: 0,
        layout: root.verticalLayout,
      }),
    );

    const data = [
      {
        descripcion: 'Morosidad', // morosidad
        cumplido: 0.8,
        pendiente: 0.2,
      },
      {
        descripcion: 'Licitaciones', // licitaciones
        cumplido: 0.4,
        pendiente: 0.6,
      },
      {
        descripcion: 'Gestion credito', // gestion credito
        cumplido: 0.25,
        pendiente: 0.75,
      },
    ];

    // Create axes
    const xRenderer = am5xy.AxisRendererX.new(root, {
      minorGridEnabled: true,
    });

    const xAxis = chart.xAxes.push(
      am5xy.CategoryAxis.new(root, {
        categoryField: 'descripcion',
        renderer: xRenderer,
        tooltip: am5.Tooltip.new(root, {}),
      }),
    );

    xRenderer.grid.template.setAll({
      location: 1,
    });

    xAxis.data.setAll(data);

    const yAxis = chart.yAxes.push(
      am5xy.ValueAxis.new(root, {
        min: 0,
        max: 100,
        // eslint-disable-next-line quotes
        numberFormat: `#'%'`,
        strictMinMax: true,
        calculateTotals: true,
        renderer: am5xy.AxisRendererY.new(root, {
          strokeOpacity: 0.1,
        }),
      }),
    );

    const legend = chart.children.push(
      am5.Legend.new(root, {
        centerX: am5.p50,
        x: am5.p50,
      }),
    );
    // Add scrollbar
    chart.set(
      'scrollbarX',
      am5.Scrollbar.new(root, {
        orientation: 'horizontal',
      }),
    );

    function makeSeries(name, fieldName) {
      // console.log(name, fieldName);

      const series = chart.series.push(
        am5xy.ColumnSeries.new(root, {
          name,
          stacked: true,
          xAxis,
          yAxis,
          valueYField: fieldName,
          valueYShow: 'valueYTotalPercent',
          categoryXField: 'descripcion',
          fill:
            fieldName !== 'cumplido'
              ? am5.color(0xb30000)
              : am5.color(0x50b300),
        }),
      );

      series.columns.template.setAll({
        // texto en columna
        // eslint-disable-next-line quotes
        tooltipText: `{name}, {categoryX}: {valueYTotalPercent.formatNumber('#.#')}%`,
        tooltipY: am5.percent(10),
      });
      series.data.setAll(data);

      // Make stuff animate on load
      // https://www.amcharts.com/docs/v5/concepts/animations/
      series.appear();

      series.bullets.push(() =>
        am5.Bullet.new(root, {
          sprite: am5.Label.new(root, {
            // eslint-disable-next-line quotes
            text: `{valueYTotalPercent.formatNumber('#.#')}% `,
            fill: root.interfaceColors.get('alternativeText'),
            centerY: am5.p50,
            centerX: am5.p50,
            populateText: true,
          }),
        }),
      );

      xAxis.get('renderer').labels.template.setAll({
        fill: root.interfaceColors.get('alternativeText'),
      });

      xAxis.setAll({
        background: am5.Rectangle.new(root, {
          fill: root.interfaceColors.get('alternativeBackground'),
          fillOpacity: 0.7,
        }),
      });

      legend.data.push(series);
    }

    makeSeries('CUMPLIDO', 'cumplido');
    makeSeries('PENDIENTE', 'pendiente');

    // chart.appear(1000, 100);
    chart.set('cursor', am5xy.XYCursor.new(root, {}));
    // eslint-disable-next-line consistent-return
    return () => {
      root.dispose();
    };
  }, []);

  return (
    <Grid
      id={idGrid}
      right={0}
      sx={{
        width: { sm: 'Calc(100% - 5%)' },
        height: '800px',
      }}
    />
  );
};

ChartRoot.propTypes = {
  idGrid: PropTypes.string.isRequired,
};
