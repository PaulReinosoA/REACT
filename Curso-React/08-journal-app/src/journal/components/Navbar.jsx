import { AppBar, Grid, IconButton, Toolbar, Typography } from '@mui/material';
import {
  BarChartOutlined,
  LogoutOutlined,
  MenuOutlined,
  NoteAddOutlined,
  TableChartOutlined,
} from '@mui/icons-material';
import { PropTypes } from 'prop-types';
import { useNavigate } from 'react-router';
import { useDispatch } from 'react-redux';
import { startLogout } from '../../store/auth/thunks';

export const Navbar = ({ drawerWidth = 240 }) => {
  const dispatch = useDispatch();
  const navigate = useNavigate(); // obtenemos la navegacion
  const onLogout = () => {
    // console.log('onLogout');
    dispatch(startLogout());
  };
  const onLoadChar = () => {
    navigate('/chart');
  };
  const onLoadTable = () => {
    navigate('/dataGrid');
  };
  const onLoadNote = () => {
    navigate('/');
  };

  return (
    <AppBar
      position="fixed"
      sx={{
        width: { sm: `calc(100% - ${drawerWidth}px)` },
        ml: { sm: `${drawerWidth}px` },
      }}
    >
      <Toolbar>
        <IconButton
          color="inherit"
          edge="start"
          sx={{ mr: 2, display: { sm: 'none' } }}
        >
          <MenuOutlined />
        </IconButton>

        <Grid
          container
          direction="row"
          justifyContent="space-between"
          alignItems="center"
        >
          <Typography variant="h6" noWrap component="div">
            {' '}
            JournalApp{' '}
          </Typography>

          <IconButton color="inherit" onClick={onLoadTable}>
            <TableChartOutlined />
          </IconButton>

          <IconButton color="inherit" onClick={onLoadChar}>
            <BarChartOutlined />
          </IconButton>

          <IconButton color="inherit" onClick={onLoadNote}>
            <NoteAddOutlined />
          </IconButton>

          <IconButton color="error" onClick={onLogout}>
            <LogoutOutlined />
          </IconButton>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};

Navbar.propTypes = {
  drawerWidth: PropTypes.number.isRequired,
};
