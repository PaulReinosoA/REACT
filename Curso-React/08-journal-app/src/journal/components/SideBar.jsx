import { Box, Divider, Drawer, List, Toolbar, Typography } from '@mui/material';
import { PropTypes } from 'prop-types';
import { useSelector } from 'react-redux';
import { SideBarItem } from './SideBarItem';

export const SideBar = ({ isActive, toggleSidebar, drawerWidth }) => {
  const { displayName } = useSelector((state) => state.auth);
  const { notes } = useSelector((state) => state.journal);

  return (
    <Box
      component="nav"
      // sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
    >
      <Drawer
        onClick={toggleSidebar}
        variant="temporary" // temporary/permanent
        open={isActive}
        sx={{
          display: { xs: 'block' },
          '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth },
        }}
      >
        <Toolbar>
          <Typography variant="h6" noWrap component="div">
            {displayName}
          </Typography>
        </Toolbar>
        <Divider />

        <List>
          {notes.map((note) => (
            <SideBarItem
              key={note.id}
              title={note.title}
              body={note.body}
              date={note.date}
              id={note.id}
              imageURL={note.imageURL === undefined ? [] : note.imageURL}
            />
          ))}
        </List>
      </Drawer>
    </Box>
  );
};

SideBar.propTypes = {
  drawerWidth: PropTypes.number.isRequired,
  isActive: PropTypes.bool.isRequired,
  toggleSidebar: PropTypes.func.isRequired,
};
