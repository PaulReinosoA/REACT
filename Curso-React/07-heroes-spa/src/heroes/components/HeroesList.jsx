// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
import { useMemo } from 'react';
import { getHeroesByPublisher } from '../helpers';
import { HeroeCard } from './HeroeCard';

export const HeroesList = ({ publisher }) => {
  const heroes = useMemo(() => getHeroesByPublisher(publisher), [publisher]);
  return (
    <>
      <div className="row row-cols-1 row-cols-md-3 g-3">
        {heroes.map((h) => (
          // eslint-disable-next-line react/jsx-props-no-spreading
          <HeroeCard key={h.id} {...h} />
          /* <li key={h.id}>{h.superhero}</li> */
        ))}
      </div>
    </>
  );
};

HeroesList.propTypes = {
  publisher: PropTypes.string.isRequired,
};
