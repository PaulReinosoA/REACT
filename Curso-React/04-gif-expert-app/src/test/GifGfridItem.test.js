import { render } from '@testing-library/react';
import { GifGfridItem } from '../components/GifGfridItem';

describe('Pruebas sobre GifGfridItem', () => {
  /*
TAREA:
1.AÃ±adir propTypes,
a.title obligatorio
b.url obligatorio
2.Evaluar snapshot
*/
  const url = 'https://bleach/bleach.jpg';
  const title = 'bleach';

  test('debe de mostarr el valores iniciales del hook', () => {
    // const GifItem =render(<GifGfridItem />)
    const { container } = render(<GifGfridItem title={title} url={url} />);
    console.log(container);
    expect(container).toBeDefined();
  });

  test('debe de hacer match con el snapshot ', () => {
    const { container } = render(<GifGfridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
});
