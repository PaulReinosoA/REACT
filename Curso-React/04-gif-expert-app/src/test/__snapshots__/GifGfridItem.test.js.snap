// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifGfridItem debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="bleach"
      src="https://bleach/bleach.jpg"
    />
    <p>
      bleach
    </p>
  </div>
</div>
`;
